### PR TITLE
Spline / Movement / WayPoints

### PR DESCRIPTION
**Pull request by DankoDJ**
Cleare most waypoint data. (new data needed - spawns)
Update MovementInfo. 
Update Splineflags. (resolved glitching for creatures on wp path)

### CODE REVIEW

**Comment by DankoDJ** [comment]
@Zyres: can you review this pr?

File: sql/world_updates/2016-09-11_01_waypoint_copy.sql [hunk 1]
```diff
@@ -0,0 +1,12 @@
+--
+-- Copy wp table and remove most data in our table
+--
+CREATE TABLE `creature_waypoints_copy` LIKE `creature_waypoints`; 
+INSERT `creature_waypoints_copy` SELECT * FROM `creature_waypoints`;
+
+DELETE FROM `creature_waypoints` WHERE `spawnid` <> 24253;
+
+--
+-- Records of spell_required
```
Comment (Zyres) [inline sql/world_updates/2016-09-11_01_waypoint_copy.sql]: wrong title

File: src/world/Movement/Spline/MovementSplineDefines.hpp [hunk 2]
```diff
@@ -21,29 +21,35 @@ namespace Movement
...
-            SPLINEFLAG_UNKNOWN4 = 0x00400000,
-            SPLINEFLAG_UNKNOWN5 = 0x00800000,
-            SPLINEFLAG_UNKNOWN6 = 0x01000000,
-            SPLINEFLAG_UNKNOWN7 = 0x02000000,
-            SPLINEFLAG_UNKNOWN8 = 0x04000000,
-            SPLINEFLAG_UNKNOWN9 = 0x08000000,
-            SPLINEFLAG_UNKNOWN10 = 0x10000000,
-            SPLINEFLAG_UNKNOWN11 = 0x20000000,
-            SPLINEFLAG_UNKNOWN12 = 0x40000000
+            SPLINEFLAG_UNK0 = 0x00000008,
```
Comment (Zyres) [inline src/world/Movement/Spline/MovementSplineDefines.hpp]: alignment

File: src/world/Objects/Object.cpp [hunk 3]
```diff
@@ -841,9 +840,9 @@ void Object::_BuildValuesUpdate(ByteBuffer* data, UpdateMask* updateMask, Player
 // This is not called!
 void Unit::BuildHeartBeatMsg(WorldPacket* data)
 {
-    data->Initialize(MSG_MOVE_HEARTBEAT, 32);
+    /*data->Initialize(MSG_MOVE_HEARTBEAT, 32);
```
Comment (Zyres) [inline src/world/Objects/Object.cpp]: do we need this opcode?

File: src/world/Objects/Object.h [hunk 4]
```diff
@@ -188,6 +188,18 @@ struct TransporterInfo
     }
 };
 
+static float NormalizeOrientation(float o)
```
Comment (Zyres) [inline src/world/Objects/Object.h]: duplicate... see LocationVector

File: src/world/Server/Packets/Handlers/MiscHandler.cpp [hunk 5]
```diff
@@ -242,402 +242,402 @@ void WorldSession::HandleAutostoreLootItemOpcode(WorldPacket& recv_data)
...
+                //check for locktypes
+
+                bool despawn = false;
+                if (pGO->GetGameObjectProperties()->chest.consumable == 1)
+                    despawn = true;
+
+                auto pLock = sLockStore.LookupEntry(pGO->GetGameObjectProperties()->chest.lock_id);
+                if (pLock != nullptr)
+                {
+                    for (uint32 i = 0; i < LOCK_NUM_CASES; i++)
```
Comment (Zyres) [inline src/world/Server/Packets/Handlers/MiscHandler.cpp]: ++i

File: src/world/Server/Packets/Handlers/MiscHandler.cpp [hunk 6]
```diff
@@ -242,402 +242,402 @@ void WorldSession::HandleAutostoreLootItemOpcode(WorldPacket& recv_data)
...
+                                }
+                                pGO->Despawn(0, sQuestMgr.GetGameObjectLootQuest(pGO->GetEntry()) ? 180000 + (RandomUInt(180000)) : (IS_INSTANCE(pGO->GetMapId()) ? 0 : 900000 + (RandomUInt(600000))));
+                                return;
+                            }
+                        }
+                        else //other type of locks that i don't bother to split atm ;P
+                        {
+                            if (pLGO->HasLoot())
+                            {
+                                pGO->SetState(1);
```
Comment (Zyres) [inline src/world/Server/Packets/Handlers/MiscHandler.cpp]: magic number

File: src/world/Server/Packets/Handlers/MiscHandler.cpp [hunk 7]
```diff
@@ -2287,234 +2287,234 @@ void WorldSession::HandleAcknowledgementOpcodes(WorldPacket& recv_data)
...
-//    li->PlayerRolled(_player, choice);
-//}
+    //	uint8 slot = 0;
+    uint32 itemid = 0;
+    uint32 amt = 1;
+    uint8 error = 0;
+    SlotResult slotresult;
+
+    Creature* pCreature = NULL;
+    GameObject* pGameObject = NULL; //cebernic added it
```
Comment (Zyres) [inline src/world/Server/Packets/Handlers/MiscHandler.cpp]: Eeeeek.... NULL argggghhhh

File: src/world/Server/Packets/Handlers/MiscHandler.cpp [hunk 8]
```diff
@@ -2287,234 +2287,234 @@ void WorldSession::HandleAcknowledgementOpcodes(WorldPacket& recv_data)
...
+        return;
+    if (pCreature)
+        pObj = pCreature;
+    else
+        pObj = pGameObject;
+
+    if (slotid >= pLoot->items.size())
+    {
+        LOG_DEBUG("AutoLootItem: Player %s might be using a hack! (slot %d, size %d)",
+                  GetPlayer()->GetName(), slotid, pLoot->items.size());
```
Comment (Zyres) [inline src/world/Server/Packets/Handlers/MiscHandler.cpp]: Log.Debug

File: src/world/Server/Packets/Handlers/MovementHandler.cpp [hunk 9]
```diff
@@ -177,31 +177,32 @@ static MovementFlagName MoveFlagsToNames[] =
...
     {
         return;
     }
 
     /************************************************************************/
     /* Clear standing state to stand.				                        */
     /************************************************************************/
-    if (recv_data.GetOpcode() == CMSG_MOVE_START_FORWARD)
-        _player->SetStandState(STANDSTATE_STAND);
+    if (opcode == CMSG_MOVE_START_FORWARD)
```
Comment (Zyres) [inline src/world/Server/Packets/Handlers/MovementHandler.cpp]: What happen if a player turn left or move backward?
Is this really the correct place to set standstate?

File: src/world/Server/Packets/Handlers/MovementHandler.cpp [hunk 10]
```diff
@@ -210,75 +211,106 @@ void WorldSession::HandleMovementOpcodes(WorldPacket& recv_data)
...
+    //    if (!(movement_info.GetMovementFlags() & MOVEFLAG_FALLING))
+    //        mover->z_axisposition = movement_info.pos.m_positionZ;
+    //}
 
     /************************************************************************/
     /* Breathing                                                            */
     /************************************************************************/
-    _player->HandleBreathing(movementInfo, this);
-
+    mover->HandleBreathing(movementInfo, this);
```
Comment (Zyres) [inline src/world/Server/Packets/Handlers/MovementHandler.cpp]: called without a function... see gew lines above

File: src/world/Storage/MySQLDataStore.cpp [hunk 11]
```diff
@@ -524,10 +524,28 @@ void MySQLDataStore::LoadCreaturePropertiesTable()
...
+                {
+                    DBC::Structures::CreatureDisplayInfoEntry const* dbc_display = sCreatureDisplayInfoStore.LookupEntry(display_info[i]);
+                    if (!dbc_display)
+                    {
+                        Log.Error("MySQLDataLoads", "Creature %u has nonexistent modelId %u (%u). To prevent client crash this model is set to 0", entry, i + 1, display_info[i]);
+                        display_info[i] = 0;
+                    }
+                }
+            }
+            creatureProperties.Male_DisplayID = display_info[0];
```
Comment (Zyres) [inline src/world/Storage/MySQLDataStore.cpp]: Should we change these different naming to an array?
On 3.3.5 too?

File: src/world/Units/Creatures/AIInterface.cpp [hunk 12]
```diff
@@ -4439,7 +4439,7 @@ void AIInterface::MoveKnockback(float x, float y, float z, float horizontal, flo
     AddSpline(x, y, z);
 
     m_Unit->m_movementManager.m_spline.GetSplineFlags()->m_splineFlagsRaw.trajectory = true;
-    m_Unit->m_movementManager.m_spline.GetSplineFlags()->m_splineFlagsRaw.knockback = true;
+    m_Unit->m_movementManager.m_spline.GetSplineFlags()->m_splineFlagsRaw.falling = true;
```
Comment (Zyres) [inline src/world/Units/Creatures/AIInterface.cpp]: Spline falling is not equal knockback...

File: src/world/Units/Players/Player.cpp [hunk 13]
```diff
@@ -7256,11 +7258,19 @@ void Player::_Relocate(uint32 mapid, const LocationVector & v, bool sendpending,
...
     if (sendpending && mapid != m_mapId && force_new_world)
     {
-        data.SetOpcode(SMSG_TRANSFER_PENDING);
-
-        data << mapid;
+        WorldPacket data(SMSG_TRANSFER_PENDING, (4 + 4 + 4));
+        data.writeBit(0);       // unknown
+        if (m_transporter)
+        {
+            data.writeBit(1);   // has transport
```
Comment (Zyres) [inline src/world/Units/Players/Player.cpp]: true/false instead o 1/0

File: src/world/Units/Unit.h [hunk 14]
```diff
@@ -111,6 +111,7 @@ enum MovementFlags
     MOVEFLAG_FALLING_SLOW = 0x08000000,   //old MOVEFLAG_FEATHER_FALL //old MOVEFLAG_FREE_FALLING //old MOVEFLAG_SPLINE_ENABLED
     MOVEFLAG_HOVER = 0x10000000,   //old MOVEFLAG_LEVITATE
     MOVEFLAG_NO_COLLISION = 0x20000000,   //old MOVEFLAG_NO_COLLISION //old MOVEFLAG_LOCAL
+    MOVEFLAG_TRANSPORT = 0x40000000,
```
Comment (Zyres) [inline src/world/Units/Unit.h]: alinment, we do not need the comment about the old value.

File: src/world/Units/Unit.h [hunk 15]
```diff
@@ -146,129 +147,129 @@ enum MovementFlags2
...
+        JumpInfo const& GetJumpInfo() const { return jump; }
+        StatusInfo const& GetStatusInfo() const { return si; }
+        float GetSplineElevation() const { return splineElevation; }
+        float GetPitch() const { return s_pitch; }
+
+    private:
+        // common
+        ObjectGuid guid;
+        ObjectGuid guid2;
+        uint32   moveFlags;                                 // see enum MovementFlags
```
Comment (Zyres) [inline src/world/Units/Unit.h]: one space.

File: src/world/Units/Unit.h [hunk 16]
```diff
@@ -1425,24 +1426,24 @@ class SERVER_DECL Unit : public Object
 
         void BuildHeartBeatMsg(WorldPacket* data);
 
-        void BuildMovementPacket(ByteBuffer *data) const;
+        /*void BuildMovementPacket(ByteBuffer *data) const;
```
Comment (Zyres) [inline src/world/Units/Unit.h]: Remove this, we should not use this anymore

**Pull request merged**